### PR TITLE
Overhaul sdist handling

### LINKED ISSUE #4436: Fails to install on Termux
There's a dedicated ruff package for Termux, but ruff-lsp and python-lsp-ruff fail to install for the same reason

```
❯ pip install --upgrade ruff
Collecting ruff
  Using cached ruff-0.0.267.tar.gz (1.1 MB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... error
  error: subprocess-exited-with-error

  × Preparing metadata (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [22 lines of output]
      error: failed to get `ruff` as a dependency of package `ruff_cli v0.0.267 (/data/data/com.termux/files/usr/tmp/pip-install-og631z7k/ruff_b22440ecd7e94053a309acbac4637fe9/crates/ruff_cli)`

      Caused by:
        failed to load source for dependency `ruff`

      Caused by:
        Unable to update /data/data/com.termux/files/usr/tmp/pip-install-og631z7k/ruff_b22440ecd7e94053a309acbac4637fe9/local_dependencies/ruff

      Caused by:
        failed to parse manifest at `/data/data/com.termux/files/usr/tmp/pip-install-og631z7k/ruff_b22440ecd7e94053a309acbac4637fe9/local_dependencies/ruff/Cargo.toml`

      Caused by:
        error inheriting `colored` from workspace root manifest's `workspace.dependencies.colored`

      Caused by:
        failed to find a workspace root
      💥 maturin failed
        Caused by: Cargo metadata failed. Does your c
rate compile with `cargo build`?
        Caused by: `cargo metadata` exited with an error:
      Error running maturin: Command '['maturin', 'pep517', 'write-dist-info', '--metadata-directory', '/data/data/com.termux/files/usr/tmp/pip-modern-metadata-9wm3nkcx', '--interpreter', '/data/data/com.termux/files/usr/bin/python3']' returned non-zero exit status 1.
      Checking for Rust toolchain....
      Running `maturin pep517 write-dist-info --metadata-directory /data/data/com.termux/files/usr/tmp/pip-modern-metadata-9wm3nkcx --interpreter /data/data/com.termux/files/usr/bin/python3`
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.

```

### PR DESCRIPTION
**Pull request by konstin**
This does three tings:

 * Fix out sdist by updating maturin (https://github.com/PyO3/maturin/issues/1609)
 * Add a test on release that ensures that the sdist keeps working
 * reduce the size of the sdist by excluding test fixtures:
   `maturin sdist && du -sh target/wheels/ruff-0.0.267.tar.gz`:
   Before: 1,1M
   After: 668K

Closes #4436.

### CODE REVIEW

**Comment by konstin** [comment]
I added a source distribution test in the release step. I think building the sdist on every CI run is overblown, but this will stop the release from going to pypi if the sdist is broken. We should do a beta release to test that the workflow actually passes (i could test in the PR test but it doesn't help too much if it doesn't upload binaries)

**Comment by konstin** [comment]
Manually triggered release run to test the CI changes: https://github.com/charliermarsh/ruff/actions/runs/5007542921

File: pyproject.toml [hunk 1]
```diff
@@ -49,6 +49,10 @@ manifest-path = "crates/ruff_cli/Cargo.toml"
 module-name = "ruff"
 python-source = "python"
 strip = true
+#exclude = [
+#  "crates/ruff/resources/test/fixtures/**/*",
+#  "crates/ruff/src/rules/*/snapshots/**/*"
+#]
```
Comment (charliermarsh) [inline pyproject.toml:55]: Nit: remove?

**Review by charliermarsh — APPROVED** [review]

File: pyproject.toml (hunk 1 above)
Comment (konstin) [reply pyproject.toml]: oops i didn't realize i committed the comments

**Comment by charliermarsh** [comment]
@konstin - I linked this PR to #4436 --  it closes that issue, right?

**Pull request merged**